### PR TITLE
Changes in playbooks for automation tests

### PR DESCRIPTION
**Pull request by vatsalparekh**
**What this PR does / why we need it**:
Changes in Test Playbooks to run them in automation
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Changes in playbooks to run them with automation
```

### CODE REVIEW

File: tests/playbooks/k8s_service.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-- name: Services work just like with standard pods
+- name: Expose VMI for SSH
```
Comment (mmazur) [inline tests/playbooks/k8s_service.yml]: `Expose VMI over SSH`

File: tests/playbooks/kubevirt_template.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            from: 'vm-[A-Za-z0-9]{8}'
+          - name: MEMORY_SIZE
+            description: Memory size
+            value: 2Gi
+
+    - name: Remove template test-template
+      kubevirt_template:
+        state: absent
+        name: test-template
+        namespace: default
```
Comment (mmazur) [inline tests/playbooks/kubevirt_template.yaml]: No newline at end of file.

**Comment by mmazur** [comment]
(Assuming CI passes. Let's wait for that.)

File: tests/playbooks/k8s_auth.yml [hunk 3]
```diff
@@ -15,17 +14,27 @@
...
+      register: node_facts
+    - name: set fact for the hostname
+      set_fact:
+        hostname: "{{ 'https://' + node_facts.resources[0].status['addresses'][-1]['address'] + ':8443'}}"
     - name: Log in (obtain auth token)
       k8s_auth:
-        username: developer
+        host: "{{ hostname }}"
+        username: test_admin
         password: "{{ k8s_developer_password }}"
```
Comment (pkliczewski) [inline tests/playbooks/k8s_auth.yml]: We changed the use name so why are we referring to `k8s_developer_password`? Why do we hardcode user name?

**Comment by mmazur** [comment]
Nope, didn't pass. The CI–detected syntax error in pvc img upload playbook needs looking into as well.

File: tests/playbooks/kubevirt_pvc.yml [hunk 4]
```diff
@@ -2,22 +2,25 @@
 - name: Create PVCs and have CDI do something with them
   hosts: localhost
   connection: local
+  module_defaults:
+    group/k8s:
+      namespace: default
+    kubevirt_pvc:
+      size: 100Mi
```
Comment (pkliczewski) [inline tests/playbooks/kubevirt_pvc.yml]: We set defaults but we ignore namespace. Why not to do the same?

File: tests/playbooks/k8s_auth.yml (hunk 3 above)
Comment (vatsalparekh) [reply tests/playbooks/k8s_auth.yml]: The purpose was to match the username in QE environment, the reason password was not changed was the user `test_admin` accepts any password.

File: tests/playbooks/k8s_service.yml (hunk 1 above)
Comment (vatsalparekh) [reply tests/playbooks/k8s_service.yml]: corrected

File: tests/playbooks/kubevirt_pvc.yml (hunk 4 above)
Comment (vatsalparekh) [reply tests/playbooks/kubevirt_pvc.yml]: Corrected

File: tests/playbooks/kubevirt_template.yaml (hunk 2 above)
Comment (vatsalparekh) [reply tests/playbooks/kubevirt_template.yaml]: Corrected.

**Comment by vatsalparekh** [comment]
> Nope, didn't pass. The CI–detected syntax error in pvc img upload playbook needs looking into as well.

Playbook couldn't find the host value, so the set task failed, the ocp in travis doesn't seem to have the route defined https://travis-ci.org/kubevirt/ansible-kubevirt-modules/builds/518202028#L8430

File: tests/playbooks/kubevirt_pvc.yml [hunk 5]
```diff
@@ -2,56 +2,52 @@
...
       kubevirt_pvc:
-        namespace: default
         name: pvc-demo2
         access_modes:
         - ReadWriteOnce
-        size: 20Mi
         cdi_source:
           pvc:
-            namespace: default
+
```
Comment (mmazur) [inline tests/playbooks/kubevirt_pvc.yml]: This is wrong, module defaults are for module parameters. `namespace` != `cdi_source.pvc.namespace`

**Comment by mmazur** [comment]
> Playbook couldn't find the host value, so the set task failed, the ocp in travis doesn't seem to have the route defined https://travis-ci.org/kubevirt/ansible-kubevirt-modules/builds/518202028#L8430

Right, so as things stand this code won't work unless either:

1. There's some kind of other route/ingress object in that travis env that gives access to that.
2. The playbook sets that access up itself.

I'm guessing you'll have to try gate numer 2.

As a side note, the code in CDI 1.7+ responsible for autodetecting this and putting the proxy url inside a CDIConfig object looks like this https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/controller/config-controller.go#L162 (lines 162 and 172 for monitoring Ingress and Route objects respectively).

File: tests/playbooks/kubevirt_pvc.yml (hunk 5 above)
Comment (vatsalparekh) [reply tests/playbooks/kubevirt_pvc.yml]: corrected

**Comment by vatsalparekh** [comment]
@mmazur removed the image upload one, will send a separate PR
The current CI failures are because templates not available in minikube, need #214 merged for that

**Comment by mmazur** [comment]
Ok, I did a full review of the changes last time, so if you've fixed/skipped the problematic stuff, I don't have any more notes. Let's wait for the Template thing to get finished and merged, then we can merge this.

**Comment by vatsalparekh** [comment]
Closing this one as most of this change is broken into parts and raised as separate PRs

**Pull request closed without merging**